Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/NoticestoMarinersControllerTest.cs

```diff
@@ -32,38 +34,79 @@ public void Setup()
+
+            IActionResult result = await _controller.Index();
             Assert.IsInstanceOf<ViewResult>(result);
             string actualView = ((ViewResult)result).ViewName;
             Assert.AreEqual(expectedView, actualView);
         }
 
         [Test]
-        public void WhenDailyFilesIsCalled_ThenShouldReturnsExpectedView()
+        public async Task WhenIndexPostIsCalled_ThenShouldReturnsExpectedViewAndViewData()
         {
-            const string expectedView = "~/Views/NoticesToMariners/ShowDailyFiles.cshtml";
-            IActionResult result = _controller.DailyFiles();
+            const string expectedView = "~/Views/NoticesToMariners/Index.cshtml";
+            const int year = 2022, week = 20;
+
+            A.CallTo(() => _fakeNMDataService.GetWeeklyFilesResponseModelsAsync(A<int>.Ignored, A<int>.Ignored, A<string>.Ignored)).Returns(SetResultForShowWeeklyFilesResponseModel());
+
+            IActionResult result = await _controller.Index(year, week);
             Assert.IsInstanceOf<ViewResult>(result);
-            string actualView = ((ViewResult) result).ViewName;
+
+            string actualView = ((ViewResult)result).ViewName;
+
             Assert.AreEqual(expectedView, actualView);
+            Assert.AreEqual(year, Convert.ToInt32(((ViewResult)result).ViewData["Year"]));
+            Assert.AreEqual(week, Convert.ToInt32(((ViewResult)result).ViewData["Week"]));
         }
 
         [Test]
-        public void WhenLoadYearsIsCalled_ThenShouldReturnJson()
+        public async Task WhenIndexPostIsCalledWithYearAndWeekZero_ThenShouldReturnsExpectedViewAndViewData()
         {
-            IActionResult result = _controller.LoadYears();
-            Assert.IsInstanceOf<JsonResult>(result);
+            const string expectedView = "~/Views/NoticesToMariners/Index.cshtml";
+            const int year = 0, week = 0, expectedViewCount = 2;
+
+            A.CallTo(() => _fakeNMDataService.GetWeeklyFilesResponseModelsAsync(A<int>.Ignored, A<int>.Ignored, A<string>.Ignored));
+
+            IActionResult result = await _controller.Index(year, week);
+            Assert.IsInstanceOf<ViewResult>(result);
+
+            string actualView = ((ViewResult)result).ViewName;
+
+            Assert.AreEqual(expectedView, actualView);
+            Assert.AreEqual(expectedViewCount, ((ViewResult)result).ViewData.Count);
         }
 
         [Test]
-        public void WhenLoadWeeksIsCalled_ThenShouldReturnJson()
+        public async Task WhenIndexPostIsCalledWithWeekOrYearZero_ThenShouldReturnsExpectedViewAndViewData()
         {
-            const int year = 2022;
-            IActionResult result = _controller.LoadWeeks(year);
-            Assert.IsInstanceOf<JsonResult>(result);
+            const string expectedView = "~/Views/NoticesToMariners/Index.cshtml";
+            const int year = 2022, week = 0;
```

Comment:
Please split these declarations onto separate lines:
```suggestion
            const int year = 2022;
            const int week = 0;
```

Reply:
Implemented declarations on separate lines